`gh repo fork --update` argument feature

F1nx opened the pull request:
### This pull request is inspired from #3189 & #444 with the `Bash` prototype/edition in [fork-fest](https://github.com/F1nx/fork-fest).

I have made a script that updates forks (specific ones or all of them) using the `gh` API, and I believe this could at least help get a start on the feature. It is up to you to decide how this project will be implemented with `cli`. I realize that this has been a sought-out issue by developers, and it would be nice to finally put the search to rest.

At the time of this request, it is in progress that the script is being translated and implemented in `Go`. It is also planned that a warning (even a confirmation) will be displayed that if anything "additional" is in the default branch of a repository, it will be left behind or deleted upon the upstream merge.

Any suggestions or corrections are appreciated, and tests can be run [here](https://github.com/F1nx/fork-update-test).

_Requested to be a "squash commit"._

Commit:
--update prompt(s)

begin --update argument contribution

ghost commented:
Thinking about just creating my own function for the `--update` argument in [`fork.go`](https://github.com/cli/cli/blob/trunk/pkg/cmd/repo/fork/fork.go).
It looks like `forkRun` was specifically made for *just* the clone and remote feature.

mislav commented:
> Thinking about just creating my own function for the `--update` argument in [`fork.go`](https://github.com/cli/cli/blob/trunk/pkg/cmd/repo/fork/fork.go).
> It looks like `forkRun` was specifically made for _just_ the clone and remote feature.

Yeah, maybe `repo fork --update` isn't the best place for this functionality. I could imagine this to be a separate command; possibly something like `gh repo sync-fork`

ghost commented:
@mislav how shall I revert the change(s) I made to `fork.go` then?

mislav commented:
@F1nx `git restore --source=upstream/trunk path/to/file.go` (assuming "upstream" is the git remote that points to our repo)

ghost commented:
@mislav Is there any *specific* way I should push the restoration?

mislav commented:
No, I don't think so. You simply `git add` the restored file, commit the revert, and `git push` it to this same branch.

Commit:
Update draft branch. (#1)

* Accept `--body-file` flag if `--body` is supported

* Remove unnecessary `BodyFile` field

* Actions Support Phase 1 (#2923)

* Implement first round of support for GitHub Actions

This commit adds:

gh actions
gh run list
gh run view
gh job view

as part of our first round of actions support. These commands are
unlisted and considered in beta.

* review feedback

* tests for exit status on job view

* spinner tracks io itself

* review feedback

* fix PR matching

* enable pager for job log viewing

* add more colorf functions

* add AnnotationSymbol

* hide job, run

* do not add method to api.Client

* remove useless cargo coded copypasta

* workflow list (#3245)

* add gh workflow list

* review feedback

* Allow retrying HTTP/2 uploads for release assets

The `Request.GetBody` func allows the retry mechanism to reopen the file
that's being uploaded as the request body in case the body of the
previous request has already started to be read.

Hopefully fixes the error:

    http2: Transport: cannot retry err [stream error: stream ID 1; REFUSED_STREAM]
    after Request.Body was written; define Request.GetBody to avoid this error

Ref. https://github.com/golang/net/blob/d523dce5a7f4b994f7ed0531dbe44cd8fd803e26/http2/transport.go#L554

* Refactor use of glamour to allow style overrides (#3243)

* Refactor use of glamour to allow style overrides

* leave the things the way they were and just expose the ability to set overrides

* add search feature in listing issues

* :nail_care: cleanup switching to search mode in `issue list`

* add generic search naming for issues and PRs @samcoe

* Add a unified GitHub Search query builder

Co-authored-by: Cristian Dominguez <cristiand391@users.noreply.github.com>
Co-authored-by: Mislav Marohnić <mislav@github.com>
Co-authored-by: Mislav Marohnić <hi@mislav.net>
Co-authored-by: Nate Smith <vilmibm@github.com>
Co-authored-by: Gowtham Munukutla <gowtham.m81197@gmail.com>

ghost commented:
I question how this command should be developed, as it seems convenient if the script passed commands to related `.go` files & functions rather than figuring things out manually with all these libraries included with `cli`.

However, in my attempt(s) of trying to make a good commit for this feature, I question if I should close this pull request since integrating this feature appears to be far out from my skill-set.
